loopdb/sql_store.go

```diff
@@ -403,6 +404,125 @@ func (s *BaseDB) BatchInsertUpdate(ctx context.Context,
 	})
 }
 
+// FetchUnconfirmedSweepBatches fetches all the batches from the database that
+// are not in a confirmed state.
+func (s *BaseDB) FetchUnconfirmedSweepBatches(ctx context.Context) ([]*Batch,
+	error) {
+
+	readOpts := NewSqlReadOpts()
+	var batches []*Batch
+
+	err := s.ExecTx(ctx, readOpts, func(tx *sqlc.Queries) error {
+		dbBatches, err := tx.GetUnconfirmedBatches(ctx)
+		if err != nil {
+			return err
+		}
+
+		for _, dbBatch := range dbBatches {
+			batch, err := s.convertBatchRow(dbBatch)
+			if err != nil {
+				return err
+			}
+
+			batches = append(batches, &batch)
+		}
+
+		return nil
+	})
+	if err != nil {
+		return nil, err
+	}
+
+	return batches, nil
+}
+
+// InsertSweepBatch inserts a batch into the database, returning the id of the
+// inserted batch.
+func (s *BaseDB) InsertSweepBatch(ctx context.Context, batch *Batch) (int32,
+	error) {
+
+	writeOpts := &SqliteTxOptions{}
+	var id int32
+	err := s.ExecTx(ctx, writeOpts, func(tx *sqlc.Queries) error {
+		args := s.BatchToInsertArgs(*batch)
+
+		var err error
+		id, err = tx.InsertBatch(ctx, args)
+		return err
+	})
+
+	if err != nil {
+		return 0, err
+	}
+
+	return id, nil
+}
+
+// UpdateSweepBatch updates a batch in the database.
+func (s *BaseDB) UpdateSweepBatch(ctx context.Context, batch *Batch) error {
+	writeOpts := &SqliteTxOptions{}
+	return s.ExecTx(ctx, writeOpts, func(tx *sqlc.Queries) error {
+		args := s.BatchToUpdateArgs(*batch)
+
+		return tx.UpdateBatch(ctx, args)
+	})
+}
+
+// ConfirmBatch confirms a batch by setting the state to confirmed.
+func (s *BaseDB) ConfirmBatch(ctx context.Context, id int32) error {
+	writeOpts := &SqliteTxOptions{}
+
+	return s.ExecTx(ctx, writeOpts, func(tx *sqlc.Queries) error {
+		return tx.ConfirmBatch(ctx, id)
+	})
+}
+
+// FetchBatchSweeps fetches all the sweeps that are part a batch.
+func (s *BaseDB) FetchBatchSweeps(ctx context.Context, id int32) (
+	[]*Sweep, error) {
+
+	readOpts := NewSqlReadOpts()
+	var sweeps []*Sweep
+
+	err := s.ExecTx(ctx, readOpts, func(tx *sqlc.Queries) error {
+		dbSweeps, err := tx.GetBatchSweeps(ctx, id)
+		if err != nil {
+			return err
+		}
+
+		for _, dbSweep := range dbSweeps {
+			sweep, err := s.ConvertSweepRow(ctx, dbSweep)
```

Comment:
I think it'd be much more readable if you'd fetch the updates here, and then make `ConvertSweepRow` a private helper function since it does not depend on anything.